encfs/BlockFileIO.cpp

```diff
@@ -191,7 +195,9 @@ bool BlockFileIO::write(const IORequest &req) {
   if (req.offset > fileSize) {
     // extend file first to fill hole with 0's..
     const bool forceWrite = false;
-    padFile(fileSize, req.offset, forceWrite);
+    int res = padFile(fileSize, req.offset, forceWrite);
+    if (res)
```

Comment:
Explicitly checking "res < 0" would help, since otherwise I have to know which functions return positive values or never greater than 0..